cmake/BLTOptions.cmake

```diff
@@ -58,6 +58,10 @@ mark_as_advanced(ENABLE_CLANG_CUDA)
 set(BLT_CLANG_CUDA_ARCH "sm_30" CACHE STRING "Compute architecture to use when generating CUDA code with Clang")
 mark_as_advanced(BLT_CLANG_CUDA_ARCH)
 option(ENABLE_HIP         "Enable HIP support" OFF)
+option(ENABLE_CLANG_HIP   "Enable Clang's native HIP support" OFF)
```

Comment:
Would it make sense to use a `cmake_dependent_option` here (with ENABLE_HIP)?

https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html

Though I suppose we don't do that for `ENABLE_CLANG_CUDA`...

Reply:
This makes sense to me, since both the `ENABLE_CLANG_X` options are ignored if the primary isn't set. If I make the change I'd probably want to change the `CLANG_CUDA` route too, just to keep things in sync.

I'm happy to make the changes, unless someone can think of a client this would break.

Reply:
We should make sure to add these these to the user docs.

Reply:
If `ENABLE_CLANG_HIP` depends on `ENABLE_HIP`, that sounds like the ideal case for a `cmake_dependent_option`

Reply:
Implemented dependent options for `CLANG_CUDA` and `CLANG_HIP`